### ISSUE TITLE
Round end roads are rendered over cycleway

### ISSUE THREAD

**Issue by Florimondable** [cyclosm/cyclosm-cartocss-style#154]
Petit pont or Pont St Michel, Paris
![Screenshot_20190908-090414](https://user-images.githubusercontent.com/47089717/64484836-521a1580-d218-11e9-8d4c-f44fb6843fd1.png)

**Comment by Bibi56** [comment]
This is probably valid for any wire display (foot ways here : https://www.cyclosm.org/#map=19/47.79712/-3.48149/cyclosm, https://www.cyclosm.org/#map=18/47.79712/-3.48149/cyclosm): roads should be displayed first.

**Comment by Phyks** [comment]
Quite related issue with motorways trunk as well: https://www.cyclosm.org/#map=18/47.04951/-1.63764/cyclosm.

![Screenshot_2019-09-21 CyclOSM](https://user-images.githubusercontent.com/3856586/65373632-5dd9f300-dc80-11e9-8098-fe23c7a2ad3c.png)

**Comment by Bibi56** [comment]
@Phyks You're right, the road network should be displayed in reverse importance order, motorways first, path last.

**Comment by Florimondable** [comment]
> @Phyks You're right, the road network should be displayed in reverse importance order, motorways first, path last.

Well no, as we do a cycle map, cycleway must be over other roads :).
And we did that on purpose, we had the case with that cycleway hided by the primary road next to it
https://www.cyclosm.org/#map=19/48.84799/2.27604/cyclosm

Though other roads (may be not path) should be render from primary to track)

**Comment by Bibi56** [comment]
@Florimondable 
> Well no, as we do a cycle map, cycleway must be over other roads :).
That's why you have to display the other network first to display cycle ways on top.

**Comment by Bibi56** [comment]
> Quick fix could be to remove the round ending for bridges?

Sounds good to me. I don't see any side effect.

**Comment by Florimondable** [comment]
I think there is possible side effects, but worth a try.

**Comment by Phyks** [comment]
Not easy to fix, no `line-cap: round` is really horrible:

![2019-10-02-172722](https://user-images.githubusercontent.com/3856586/66058176-2f8acc00-e53a-11e9-87dd-ab23467a7bfe.png)

Openstreetmap-carto has this quite enigmatic comment https://github.com/gravitystorm/openstreetmap-carto/blob/master/roads.mss#L1107-L1108 which we might dig into. They don't have this issue in the render, so they probably do it right https://www.openstreetmap.org/#map=18/48.85327/2.34692&layers=N.

**Comment by Florimondable** [comment]
Because path are rendered under main highway osm.org don't have the issue we see in your try.

**Comment by Florimondable** [comment]
I think regular path (not bicycle=designated) can be rendered under main highway in CyclOSM too.

**Comment by Bibi56** [comment]
You seem to render bridges before cycle ways.

If you render bridges after, it _may_ be correct: render normal network (outside bridges) then render bridges but this includes cycleways on bridges, it may difficult to do.

I get one part of the comment (if you render something with transparency several times it changes the transparency, that's why it "glows"), not the second part (why changing line-cap forces multiple rendering, I guess it has to do with the Mapnik rendering engine, maybe the issue is solved now).

If you look at the cycling map on osm.org, https://www.openstreetmap.org/#map=18/48.85327/2.34625&layers=CN, they seem to display the cycleways last with transparency.
That doesn't take into account that the way is under the bridge but seems to be a good compromise.

**Comment by Phyks** [comment]
> You seem to render bridges before cycle ways.

We should not, bridges are in [this layer](https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/project.mml#L721) while the regular cycleways are on [this layer](https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/project.mml#L562) below.

> If you look at the cycling map on osm.org, https://www.openstreetmap.org/#map=18/48.85327/2.34625&layers=CN, they seem to display the cycleways last with transparency.
That doesn't take into account that the way is under the bridge but seems to be a good compromise.

They however seem to have similar issues on the next bridge :) https://www.openstreetmap.org/#map=18/48.85390/2.34429&layers=CN

**Comment by Florimondable** [comment]
link to #192
